### PR TITLE
Refacoring system tests

### PR DESCRIPTION
**Pull request by dneukirchen**
* Implement naming conventions for steps, pages and cests
* Implement inheritance for page objects (steps and helpers will follow)
* Enable JoomlaDb to test against database instead of performance heavy browser actions
* Code style fixes
* Use Dependency injection instead of newing up steps
* Delete unused data files
* Refactor com_content tests (needs some work)
* Start removing logic that does not belong in page objects. we use page objects only for locators!
* Unify tests/steps/helpers

### CODE REVIEW

File: src/_support/Helper/JoomlaDb.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ public function haveInDatabase($table, array $data)
 	}
 
 	/**
-	 * Find an entry in the database
+	 * Ass an entry in the database
```
Comment (yvesh) [inline src/_support/Helper/JoomlaDb.php]: Ass really? :D As

Comment (dneukirchen) [reply src/_support/Helper/JoomlaDb.php]: :D :D :D

File: src/acceptance/administrator/components/com_content/ContentListCest.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	// Clear the search of articles
+
+	// Sorting articles
+	// id
+
+	// Filter articles
+	// status
+	// category
+	// user
+	// tag
```
Comment (yvesh) [inline src/acceptance/administrator/components/com_content/ContentListCest.php:147]: Can you add the todos to the project board ;-)

Comment (dneukirchen) [reply src/acceptance/administrator/components/com_content/ContentListCest.php:147]: I already have follow up branch ready.

**Comment by yvesh** [comment]
Thank you @dneukirchen !!!

**Pull request merged**